Fix build on ZUUL

oshogbo opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/intel/ScaffoldingForGramine/16)
<!-- Reviewable:end -->

oshogbo committed:
Fix names of directories and jobs

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
All tests require aesm

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Consolidate apt-get commands and resolve hadolint warnings

This commit consolidates two apt-get commands to ensure that
the `apt-get update` command is synchronized with
`apt-get install`. This prevents potential issues where Docker
caches the layer of each RUN command, potentially using outdated
distribution files with `apt-get install`.

Additionally, this commit addresses hadolint warnings.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Define the image variable in the global scope

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Delete the apt-get lists after installing something

Also add this check as ignored by hadolint.
Hadolint requires after each apt-get install this line,
but I don't want to force user to remember about that.
We have this in main Dockerfile, which means that it will
be added to all Dockerfiles automagically.

Requested by hadolint.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Ignore Multiple RUN Cases

We are keeping commands in separate RUN statements to avoid
requiring the user to manually run them, as our Dockerfile
is seperated in multiple sections.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

oshogbo committed:
Extend Expect timeout to 90 seconds

The default timeout for Expect is 10 seconds, which may not suffice
as the Docker project build time can exceed this duration.
To accommodate longer build times, the timeout has been extended to
90 seconds.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

woju reviewed (approved):
> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/16)* status: all files reviewed, 1 unresolved discussion (waiting on @oshogbo)

___
*[`graminescaffolding/templates/Dockerfile` line 25 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/16#-NfjjDOa3fLty43Fwf_z:-NfjjDOa3fLty43Fwfa-:bjcf0v1) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/8c3edf144ba2aae03b9eda49a825a9a05897e348/graminescaffolding/templates/Dockerfile#L25)):*
> ```Dockerfile
> {# dockerfile_generate_manifest -#}
> 
> # hadolint ignore=DL3059
> ```

DL3059 should probably be ignored globally at this point. Not blocking, but I would do it somewhere project-wide.


<!-- Sent from Reviewable.io -->

oshogbo committed:
ci: make DL3059 hadolint rule ignored globally

This Rule catches multiple consecutive RUN instructions. Seems
that we can't avoid this issue.

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

woju reviewed (approved):
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/16)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @oshogbo)



<!-- Sent from Reviewable.io -->
